File: src/main/kotlin/com/codingdojos/api/model/Dojo.kt
package com.codingdojos.api.model

import java.time.LocalDate
import java.util.*

data class Dojo(
        val id: String = UUID.randomUUID().toString(),
        val label: String,
        val status: DojoStatus = DojoStatus.Proposed,
        val subject: Subject? = null,
        val location: String? = null,
        val date: LocalDate? = null,
        val poll: DatePoll? = null
)

File: client/src/services/model/Dojo.ts
import Subject from './Subject';
import DatePoll from './DatePoll';

export default class Dojo {
  public id: string | null;
  public label: string;
  public status: string;
  public subject: Subject | null;;
  public location: string | null;
  public date: string | null;
  public poll: DatePoll | null;
}
